### PR TITLE
slof_user_interface: Add slof_user_interface case for SLOF

### PR DESCRIPTION
**Pull request by YongxueHong**
slof_user_interface is that verify SLOF info by user interface.

ID: 1646846
Signed-off-by: Yongxue Hong <yhong@redhat.com>

### CODE REVIEW

**Comment by YongxueHong** [comment]
Depends on #1531

File: qemu/tests/slof_user_interface.py [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+                         % '->'.join(option.split()))
+
+            dev_type, hba_type, child_bus, addr = tuple(option.split())
+            addr = re.sub(r'^0x0?', '', addr)
+            pattern = re.compile(r'(\d+)\)\s+%s(\d+)?\s+:\s+/%s(\S+)?/%s@%s' %
+                                 (dev_type, hba_type, child_bus, addr), re.M)
+            searched = pattern.search(menu_info)
+            if not searched:
+                test.fail('No such item(%s) in boot menu list.' %
+                          '->'.join(option.split()))
```
Comment (wangxianxian) [inline qemu/tests/slof_user_interface.py]: if pass, it's better to give related pass log.

**Comment by wangxianxian** [comment]
slof_user_interface.f12: PASS
slof_user_interface.boot: PASS
slof_user_interface.reset_all: PASS

**Comment by YongxueHong** [comment]
@hereischen  @lolyu  pls help me review, thanks.

File: qemu/tests/slof_user_interface.py [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+    STOP, F12 = range(2)
+    enter_key = {STOP: 's', F12: 'f12'}
+
+    def _create_iso():
+        """ Create iso with one file on it. """
+        if params.get("create_cd", 'no') == 'yes':
+            logging.info("creating iso.")
+            iso_name = params.get("cdrom_cd1")
+            iso = utils_misc.get_path(data_dir.get_data_dir(), iso_name)
+            process.system_output(
```
Comment (hereischen) [inline qemu/tests/slof_user_interface.py]: why not call `process.system_output('dd if=/dev/urandom of=new bs=10M count=1&&mkisofs -o %s new...')` once?

File: qemu/tests/slof_user_interface.py [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+
+    def _check_menu_info(menu_info):
+        """ Check the menu info by each items. """
+        bootable_num = ''
+        for i in range(1, int(params['boot_dev_num']) + 1):
+            option = params['menu_option%d' % i]
+            logging.info('Checking the device(%s) if is included in menu list.'
+                         % '->'.join(option.split()))
+
+            dev_type, hba_type, child_bus, addr = tuple(option.split())
```
Comment (hereischen) [inline qemu/tests/slof_user_interface.py]: don't need `tuple()`
`>>> a,b,c = [1,2,3]
>>> a
1
`

File: qemu/tests/slof_user_interface.py (hunk 1 above)
Comment (YongxueHong) [reply qemu/tests/slof_user_interface.py]: As talked with xianwang, cancel this requirement.

File: qemu/tests/cfg/slof_user_interface.cfg [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+            menu_option4 = "disk vdevice v-scsi ${spapr_vscsi_reg}"
+            bootable_index = 3
+        - boot:
+            send_keys = 'boot'
+            bootindex_image1 = 0
+            start_boot = 'Trying to load'
+            end_boot = 'Successfully loaded'
+        - reset_all:
+            send_keys = 'reset-all'
+            start_reset-all = '0 >'
```
[the remark this answers is not in the document]
Comment (YongxueHong) [reply qemu/tests/cfg/slof_user_interface.cfg]: Thanks, have updated.

File: qemu/tests/slof_user_interface.py (hunk 3 above)
Comment (YongxueHong) [reply qemu/tests/slof_user_interface.py]: Thanks, have updated.

File: qemu/tests/slof_user_interface.py (hunk 2 above)
Comment (YongxueHong) [reply qemu/tests/slof_user_interface.py]: Thanks, have updated.

File: qemu/tests/slof_user_interface.py [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+    main_tests = {'f12':  _f12_user_interface_test,
+                  'boot': _load_user_interface_test,
+                  'reset-all': _load_user_interface_test}
+
+    try:
+        create_cd = params.get('create_cd', 'no')
+        if create_cd == 'yes':
+            _create_iso()
+
+        params['start_vm'] = 'yes'
```
Comment (hereischen) [inline qemu/tests/slof_user_interface.py]: I think you can remove `params['start_vm'] = 'yes'`

Comment (YongxueHong) [reply qemu/tests/slof_user_interface.py]: thanks, have updated.

**Comment by hereischen** [comment]
Acked-by: Haotong Chen hachen@redhat.com

**Comment by wangxianxian** [comment]
ACK

**Comment by YongxueHong** [comment]
Thanks all.

**Pull request merged**